multimodal/src/autogluon/multimodal/utils/misc.py

```diff
@@ -74,3 +75,114 @@ def path_expander(path, base_folder):
+            if (
+                last_start >= 0
+                and (not entity_group.startswith("B-"))
+                and (
+                    (entity_group.startswith("I-") and last_label[2:] == entity_group[2:])
+                    or last_label == entity_group
+                )
+                and self.sent[last_end:start].isspace()
+            ):
+                last_end = end
+            else:
+                last_start = start
+                last_end = end
+                last_label = entity_group
+
+            spans.update({last_start: (last_end, last_label)})
+        return spans
+
+    @staticmethod
+    def escape_html(text: str) -> str:
+        """Replace <, >, &, " with their HTML encoded representation. Intended to
+        prevent HTML errors in rendered displaCy markup.
+        text (str): The original text.
+        RETURNS (str): Equivalent text to be safely used within HTML.
+        """
+        text = text.replace("&", "&amp;")
+        text = text.replace("<", "&lt;")
+        text = text.replace(">", "&gt;")
+        text = text.replace('"', "&quot;")
+        return text
+
+    def html_template(self, text, label, color):
+        """
+        Generate an HTML template for the given text and its label.
+
+        Parameters
+        ----------
+        text
+            The text to be highlighted.
+        label
+            The predicted label for the given text.
+        color
+            The background color of the mark tag.
+        """
+        text = '<mark style="background-color:{}; color:white; border-radius: 1em .1em; padding: .3em;">{} \
+         <b style="background-color:white; color:black; font-size:x-small; border-radius: 0.3em .3em; padding: .1em;">{} </b> \
+         </mark>'.format(
+            color, self.escape_html(text), self.escape_html(label)
+        )
+        return text
+
+    def _repr_html_(self):
+        entities = []
+        new_sent = ""
+        last = 0
+        for key, value in self.spans.items():
+            entity_group = value[-1]
+            if entity_group.startswith("B-") or entity_group.startswith("I-"):
+                entity_group = entity_group[2:]
+            if entity_group not in self.colors:
+                self.colors.update({entity_group: "#%06X" % randint(0, 0xFFFFFF)})
+            start = key
+            new_sent += self.sent[last:start]
+            last = end = value[0]
+            entity_text = self.html_template(self.sent[start:end], entity_group, color=self.colors[entity_group])
+            new_sent += entity_text
+        new_sent += self.sent[last:]
+
+        return new_sent
+
+
+def visualize_ner(sentence, prediction):
+    """
+    Visualize the prediction of NER.
+
+    Parameters
+    ----------
+    sentence
+        The input sentence.
+    prediction
+        The NER prediction for the sentence.
+
+    Returns
+    -------
+    An NER html visualizer.
+    """
+    visualizer = NERVisualizer(prediction, sentence)
```

Comment:
A little curious about the advantages of defining class `NERVisualizer `. Is the visualization effect the same if we call a function with the same functionality as the above `_repr_html_`?

Reply:
Can we add a docstring to the `_repr_html_` so that developers would know about this interface?